src/db.c

```diff
@@ -2265,15 +2265,29 @@ int migrateGetKeys(struct redisCommand *cmd, robj **argv, int argc, getKeysResul
     num = 1;
 
     /* But check for the extended one with the KEYS option. */
-    if (argc > 6) {
+    struct {
+        char* name;
+        int skip;
+    } skiplist[] = {       
+        {"copy", 0},
+        {"replace", 0},
+        {"auth", 1},
+        {"auth2", 2},
+        {NULL, 0}
+    };
+    if (argc > 6 && sdslen(argv[3]->ptr) == 0) {
         for (i = 6; i < argc; i++) {
-            if (!strcasecmp(argv[i]->ptr,"keys") &&
-                sdslen(argv[3]->ptr) == 0)
-            {
-                first = i+1;
-                num = argc-first;
+            if (!strcasecmp(argv[i]->ptr, "keys")) {
+                first = i + 1;
+                num = argc - first;
                 break;
             }
+            for (j = 0; skiplist[j].name != NULL; j++) {
+                if (!strcasecmp(argv[i]->ptr, skiplist[j].name)) {
+                    i += skiplist[j].skip;
+                    break;
+                }
+            }
```

Comment:
don't you need this block to be executed before line 2280?

Reply:
ok maybe it's not necessary, but i think it belongs in the beginning of the loop

Reply:
I don't think it really matters. And, executing this block after line 2280 makes it break earlier when the loop reaches the "keys" keyword.